src/nsrt_learning/sampler_learning.py

```diff
@@ -220,8 +222,36 @@ def _create_sampler_data(
     datastore_idx: int
 ) -> Tuple[List[SamplerDatapoint], List[SamplerDatapoint]]:
     """Generate positive and negative data for training a sampler."""
-    positive_data = []
-    negative_data = []
+    # Populate all positive data.
+    positive_data: List[SamplerDatapoint] = []
+    for (segment, var_to_obj) in datastores[datastore_idx]:
+        # Note: it should ALWAYS be the case that the segment has
+        # an option here. If options are learned, then earlier calls
+        # to update_segment_from_option_spec() should have set the
+        # option correctly to a learned one.
+        option = segment.get_option()
+        state = segment.states[0]
+        if CFG.sampler_learning_use_goals:
+            # Right now, we're making the assumption that all data is
+            # demonstration data when we're learning samplers with goals.
+            # In the future, we may weaken this assumption.
+            assert segment.has_goal()
+            goal = segment.get_goal()
+        else:
+            goal = None
+        assert all(
```

Comment:
add a similar assertion for the effects?

Reply:
you mean assert that all the add effects hold in the `segment.states[1]`?